pkgs/applications/editors/emacs/default.nix

```diff
@@ -67,11 +61,17 @@ stdenv.mkDerivation rec {
              "--with-gif=no" "--with-tiff=no" ])
     ++ lib.optional withXwidgets "--with-xwidgets";
 
-  NIX_CFLAGS_COMPILE = lib.optionalString (stdenv.isDarwin && withX)
-    "-I${cairo.dev}/include/cairo";
-
-  preBuild = ''
+  preConfigure = lib.optionalString srcRepo ''
+    ./autogen.sh
+  '' + lib.optionalString srcLocal ''
```

Comment:
I just realized that all the tarballs already contain byte-compiled elisp files. Therefore that step was really detrimental to compilation time for the general case.